### PR TITLE
Campaign page UI improvements

### LINKED ISSUE #15: Implement - UI Play page

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
***This is a UI enhancement PR and thus should also be reviewed on the webpage in a running state in addition to the source code.***

**Summary of added content:**
- Improvements to the Campaign page
  - Adds a scroll-to-top button to make it easy to start reading the story from the beginning
    - The button only appears if there is a scroll bar, e.g., if there is a reason to scroll to the top
  - Automatically scrolls to the most recent message when the user opens a saved campaign
  - Select bar for prompt types [Do, Say, Attack]
  - Placeholders in the user input box change depending on the chosen prompt type 
  - Floating game statistics panels
    - Game information
    - Character information
  - Idle spinner when waiting for LLM responses
  - Health bar to track player HP

A campaign in a 1920x1080 resolution web browser:
image

**Relevant files for review:**
- `ChatRPG/Pages/CampaignPage.razor`
- `ChatRPG/Pages/CampaignPage.razor.cs`
- `ChatRPG/wwwroot/js/detectScrollBar.js`
- `ChatRPG/Pages/HealthBar.razor`

This pull request closes #15

### CODE REVIEW

**Review by KasperHenningsen — APPROVED** [review]
LGTM

File: ChatRPG/Pages/CampaignPage.razor.cs [hunk 1]
```diff
@@ -48,9 +69,15 @@ protected override async Task OnInitializedAsync()
 
         _campaign = await PersisterService!.LoadFromCampaignIdAsync(
             CampaignMediatorService!.UserCampaignDict[_loggedInUsername!]);
+        _npcList = _campaign.Characters.ToList();
+        _currentLocation = _campaign.Environments.LastOrDefault();
+        _mainCharacter = _campaign.Player;
         if (_campaign != null)
```
Comment (KasperHenningsen) [inline ChatRPG/Pages/CampaignPage.razor.cs:75]:
```suggestion
        if (_campaign)
```

**Review by KasperHenningsen — CHANGES_REQUESTED** [review]

File: ChatRPG/Pages/CampaignPage.razor.cs (hunk 1 above)
Comment (KarmaKamikaze) [reply ChatRPG/Pages/CampaignPage.razor.cs:75]: Hold the phone, det er sabotage! 🤣

Comment (KarmaKamikaze) [reply ChatRPG/Pages/CampaignPage.razor.cs:73]: Men helt skidt var det ikke for det fik mig til at indse at de tre tilføjede linjer (72, 73, og 74) nok burde være inde i null checket! Så tak 😄 Nu må du gerne re-review!

File: ChatRPG/Pages/CampaignPage.razor.cs [hunk 2]
```diff
@@ -48,9 +69,16 @@ protected override async Task OnInitializedAsync()
...
+
         if (_campaign != null)
         {
-            _combatMode = _campaign.CombatMode;
+            _npcList = _campaign.Characters.ToList();
+            _currentLocation = _campaign.Environments.LastOrDefault();
+            _mainCharacter = _campaign.Player;
+            if (_campaign.CombatMode)
+            {
+                _activePromptType = PromptType.Attack;
```
Comment (KasperHenningsen) [inline ChatRPG/Pages/CampaignPage.razor.cs:80]:
```suggestion
                _activePromptType = PromptType.Attack unless _campaign.Characters.ToList().first. pacifist?
```

Comment (KarmaKamikaze) [reply ChatRPG/Pages/CampaignPage.razor.cs:80]: Ruby on rails invasion?! Make it stop!

**Review by KasperHenningsen — APPROVED** [review]
Det er vel fint nok for et skoleprojekt

File: ChatRPG/Pages/CampaignPage.razor.cs (hunk 2 above)
Comment (KasperHenningsen) [reply ChatRPG/Pages/CampaignPage.razor.cs:80]: image

**Review by damniko — APPROVED** [review]
# I have reviewed
It was a long read, but it was worth the read.

# My conclusion
The changes in this PR are hereby approved with the DamStamp™ of Approval<sup>®</sup>.

**Pull request merged**